packages/frontend-2/lib/settings/composables/management.ts

```diff
@@ -89,7 +90,8 @@ export function useAddWorkspaceDomain() {
                 ],
                 discoverabilityEnabled:
                   domains.length === 0 ? true : discoverabilityEnabled,
-                domainBasedMembershipProtectionEnabled
+                domainBasedMembershipProtectionEnabled,
```

Comment:
This whole optimisticResponse block looks very hacky, with the `as Workspace` bit and readFragment(). Why do we need an optimisticResponse here at all?

Reply:
@fabis94 I think there were two parts of this, one part was added because in the UI we needed to reflect the list of updated domains, but it was not available, the other was added later on during SSO but I didn't work it that. You have any suggestions for a better approach?

Reply:
@Mikehrn Its still not clear to me why optimisticResponse was needed. All it does is return a fake response before the real one comes back. But surely the real one comes back in very short time like all other GQL mutations, so why not just use that?

optimisticResponse is used for things like like buttons where you want the button to toggle immediately, instead of having to wait for the API call to finish